IoT/Files/Configuration/IoTHubReuseConfiguration.json

```diff
@@ -0,0 +1,57 @@
+{
+  "variables": [
+    {
+      "id": "hubName",
+      "value": "{{#parseAzureResourceId inputs.iotHubId 8}}{{/parseAzureResourceId}}"
+    },
+    {
+      "id": "hubResourceGroup",
+      "value": "{{#parseAzureResourceId inputs.iotHubId 4}}{{/parseAzureResourceId}}"
+    }
+  ],
+  "assets": [
+    {
+      "id": "updateTags",
+      "type": "UpdateDeploymentTags",
+      "stage": "post",
+        "inputs": {
+            "resourceId": "/subscriptions/{{inputs.subscriptionId}}/resourceGroups/{{variables.hubResourceGroup}}/providers/Microsoft.Devices/IotHubs/{{variables.hubName}}",
+            "apiVersion": "2018-04-01",
+            "endpoint": "{{assets.ARMEndpoint_object}}"
+        }
+    }
+  ],
+  "releaseDefinition": {
+    "environments": [
+      {
+        "name": "dev",
+        "deployedResourceIds": [
+          "/subscriptions/{{{inputs.subscriptionId}}}/resourceGroups/{{{variables.hubResourceGroup}}}/providers/Microsoft.Devices/IotHubs/{{{variables.hubName}}}"
+        ],
+        "variables": {
+          "DEVOPS_IOTEDGE_REGISTRY_URL": {
+            "value": "{{{inputs.containerRegistryName}}}.azurecr.io"
+          }
+        },
+        "templateFile": "Files/ReleaseDefinitionTemplates/deployIoTEdgeWithExistingHub.json",
+        "phaseInputs": [
+          {
+            "queue": "Bishal",
+            "taskInputs": [
+              {
+                "__TaskId__": "46e4be58-730b-4389-8a2f-ea10b3e5e815",
+                "connectedServiceNameARM": "{{{assets.ARMEndpoint}}}",
+                "inlineScript": "(az extension add --name azure-cli-iot-ext && az iot hub device-identity show --device-id myEdgeDevice --hub-name {{{variables.hubName}}}) || (az iot hub device-identity create --hub-name {{{variables.hubName}}} --device-id myEdgeDevice --edge-enabled && TMP_OUTPUT=\"$(az iot hub device-identity show-connection-string --device-id myEdgeDevice --hub-name {{{variables.hubName}}})\" && RE=\"\\\"cs\\\":\\s?\\\"(.*)\\\"\" && if [[ $TMP_OUTPUT =~ $RE ]]; then CS_OUTPUT=${BASH_REMATCH[1]}; fi && echo \"##vso[task.setvariable variable=CS_OUTPUT]${CS_OUTPUT}\")"
```

Comment:
The TMP_OUTPUT and CS_OUTPUT is to get connection string and use it when creating VM. If VM is not needed, these logic can be removed

Reply:
Yes, this logic will most probably be removed. I am not yet done with exact BD and RD. Will update it in next PR.

Reply:
I have updated BD and RD templates and input overrides. You can take a look.

Reply:
I see that the create Edge device logic is also removed and release pipeline only contains one task targeting `tags.edgeDevice='true'`. Then we require user to manually tag their own existing Edge device with the key `edgeDevice`?

Reply:
I think create Edge device can be retained, but no need to output the device connection string since it was used as a parameter passed to VM
So the original script can be simplified
`(az extension add --name azure-cli-iot-ext && az iot hub device-identity show --device-id myEdgeDevice --hub-name {{{variables.hubName}}}) || (az iot hub device-identity create --hub-name {{{variables.hubName}}} --device-id myEdgeDevice --edge-enabled)`
(not tested)

Reply:
@michaeljqzq Yes, the user is expected to tag their devices manually. This is current behavior which we are targeting for internal demo.
Ultimately, we want user to specify device name or a pattern matching multiple device names or a tag. Since, it is an existing hub with devices already attached, tag is not a bad option. What would be a good productized behavior in your opinion

Reply:
@michaeljqzq right now I will go ahead with this behavior. Later based on the PM spec, we will change it.